src/Symfony/Component/Messenger/Bridge/GooglePubSub/Transport/Connection.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Bridge\GooglePubSub\Transport;
+
+use Google\Cloud\Core\Exception\GoogleException;
+use Google\Cloud\PubSub\PubSubClient;
+use Google\Cloud\PubSub\Subscription;
+use Google\Cloud\PubSub\Topic;
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+use Symfony\Component\Messenger\Exception\TransportException;
+
+/**
+ * @author Richard Henkenjohann <richardhenkenjohann@googlemail.com>
+ *
+ * @internal
+ */
+class Connection
+{
+    private const DEFAULT_OPTIONS = [
+        // PubSub client
+        'apiEndpoint' => null,
+        'projectId' => null,
+        'keyFile' => null,
+        'keyFilePath' => null,
+        'requestTimeout' => null,
+        'retries' => 3,
+        'scopes' => null,
+        'quotaProject' => null,
+
+        // PubSub topic
+        'topic' => [
+            'name' => null
+        ],
+
+        // PubSub subscription
+        'subscription' => [
+            'name' => null
+        ],
+
+        // Internal
+        'auto_setup' => true,
+    ];
```

Comment:
I don't know for others but maybe it could be a good idea to use `OptionsResolver` to define the options of this transport, this way, we can easily inspect and validate the submitted options when building it?